packages/ses/src/error/tame-v8-error-constructor.js

```diff
@@ -299,8 +301,37 @@ export const tameV8ErrorConstructor = (
     // Use concise methods to obtain named functions without constructors.
     const systemMethods = {
       prepareStackTrace(error, sst) {
-        weakmapSet(stackInfos, error, { callSites: sst });
-        return inputPrepareFn(error, safeV8SST(sst));
+        const stackInfo = {};
+        const safeCallSites = safeV8SST(sst);
+        let preparedStack;
+        try {
+          preparedStack = inputPrepareFn(error, safeCallSites);
+          if (preparedStack === safeCallSites) {
+            // Handle depd and similar prepareStackTrace which return the structured stack trace
+            stackInfo.callSites = sst;
+            return safeCallSites;
+          } else if (
+            typeof preparedStack === 'string' &&
+            preparedStack !== ''
+          ) {
+            stackInfo.stackString = stringReplace(preparedStack, /^[\n]+/, '');
+          } else {
+            // We just need to get to the `catch` clause
+            // eslint-disable-next-line no-throw-literal
+            throw undefined;
+          }
+        } catch (_err) {
+          if (errorTaming === 'unsafe') {
+            stackInfo.stackString = stackStringFromSST(error, sst);
+            preparedStack = `${errorToString(error)}${stackInfo.stackString}`;
+          } else {
+            stackInfo.callSites = sst;
```

Comment:
I had to read very carefully before I was satisfied that the original `sst` still cannot leak. That invariant seems less local now. Worth a comment?

Reply:
I see the 👍 . Resolving.